src/Illuminate/Routing/ResourceRegistrar.php

```diff
@@ -28,6 +27,13 @@ class ResourceRegistrar
      */
     protected $singletonResourceDefaults = ['show', 'edit', 'update'];
 
+    /**
+     * The default actions for a api singleton resource controller.
+     *
+     * @var string[]
+     */
+    protected $apiSingletonResourceDefaults = ['show', 'update'];
+
```

Comment:
We don't have a corresponding property for `$apiResourceDefaults`, which feels asymmetrical.

In my testing, the `apiResource` method has the same behaviour; it will register the `edit` route if explicitly provided to the `only` method.

I am curious to know how much of an issue this is. Perhaps there are situations where someone may want the `edit` route if they explicitly ask for it.

If we did want to fix this issue, perhaps a property like `$excludedApiActions = ['create', 'edit'];` might be better so that we can just remove those actions from both `apiResource` and `apiSingletonResource`.

Reply:
Hi Jess, I confess I got a bit carried away and maybe changed a bit too much...

I believe that the `edit` route should never exist in an API resource, am I wrong?

Usually I tend to use a positive approach, instead if using an exclusion approach, I think it's easier to read. I'm going to review this and remove the `$apiSingletonResourceDefaults` I should've used a less intrusive approach.